### PR TITLE
Fixed a bug in MeshComponentMap under PETSc with single thread

### PR DESCRIPTION
**Pull request by wenqing**
As titled.

One benchmark is added for this bug fixing.

### CODE REVIEW

File: NumLib/DOF/MeshComponentMap.cpp [hunk 1]
```diff
@@ -86,48 +97,29 @@ MeshComponentMap::MeshComponentMap(
...
-    _num_local_dof = _dict.size();
-
-    if (order == ComponentOrder::BY_LOCATION)
-    {
-        renumberByLocation();
-    }
+    createSerialMeshComponentMap(components, order);
 }
-#endif // end of USE_PETSC
+#endif  // end of USE_PETSC
```
Comment (wenqing) [inline NumLib/DOF/MeshComponentMap.cpp]: The changes from line 110 to line 364 are just caused by clang-format.

File: ProcessLib/RichardsMechanics/Tests.cmake [hunk 2]
```diff
@@ -182,3 +182,17 @@ AddTest(
...
+AddTest(
+    NAME Parallel_RichardsMechanics_RichardsFlow_2d_richardsflow
+    PATH RichardsMechanics
+    EXECUTABLE_ARGS RichardsFlow_2d_richardsflow.prj
+    WRAPPER mpirun
+    WRAPPER_ARGS -np 1
+    TESTER vtkdiff
+    REQUIREMENTS OGS_USE_MPI
+    DIFF_DATA
+    RichardsFlow_2d_richardsflow_pcs_0_ts_99_t_1900.000000.vtu RichardsFlow_2d_richardsflow_pcs_0_ts_99_t_1900_000000_0.vtu pressure pressure 5e-8 1e-10
```
Comment (wenqing) [inline ProcessLib/RichardsMechanics/Tests.cmake]: Use the results of the serial computation as the references.

**Comment by endJunction** [comment]
@wenqing I just split off the clang-tidy changes and dropped some unrelated changes. This way it is easier to review code.

**Comment by endJunction** [comment]
Just run the `Tests/Data/RichardsMechanics/RichardsFlow_2d_small.prj` test with ogs compiled with petsc but it wasn't successful:
```
> bin/ogs ../s/Tests/Data/RichardsMechanics/RichardsFlow_2d_small.prj
...
[0] === Time stepping at step #1 and time 1 with step size 1
...
================================================
Linear solver cg with bjacobi preconditioner
converged in 61 iterations (relative convergence criterion fulfilled).
================================================
[0] [time] Linear solver took 0.00374283 s.
[0] Convergence criterion, component 0: |dx|=2.0870e+04, |x|=6.8291e+04, |dx|/|x|=3.0560e-01
[0] Convergence criterion, component 1: |dx|=1.1547e-20, |x|=1.1857e-20, |dx|/|x|=9.7390e-01
[0] Convergence criterion, component 2: |dx|=1.0881e-11, |x|=1.0654e-08, |dx|/|x|=1.0213e-03
[0] [time] Iteration #2 took 0.594093 s.
[0] Assemble the Jacobian of RichardsMechanics for the monolithic scheme.
[0] [time] Assembly took 0.590075 s.
[0] [time] Applying Dirichlet BCs took 0.000618455 s.

Divergence because of indefinite preconditioner,
Try to run again with -pc_factor_shift_positive_definite option.
[0] [time] Linear solver took 0.000860681 s.
[0] Newton: The linear solver failed.
```
Didn't look into details though...

**Comment by wenqing** [comment]
@endJunction  The command to run that benchmark is
```
mpirun -np 1  bin/ogs ../s/Tests/Data/RichardsMechanics/RichardsFlow_2d_small.prj
```

**Comment by wenqing** [comment]
Tried. One can also run the MPI executable file as that for  serial one, e.g.
```
bin/ogs ../s/Tests/Data/RichardsMechanics/RichardsFlow_2d_small.prj
```
Under such case, the rank size is set to 1 by default.
 With that serial command, that benchmark works as well on frontend 2 of eve.

**Pull request merged**